### PR TITLE
Take 2: Modify DRS Parsing 

### PR DESCRIPTION
**Pull request by ruchim**
What used to be invalid DRS URI but *is* now a valid DRS URI:
dos://<uuid>

### CODE REVIEW

File: cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsCloudNioFileSystemProvider.scala [hunk 1]
```diff
@@ -36,18 +36,21 @@ class DrsCloudNioFileSystemProvider(rootConfig: Config,
...
      * In some cases for a URI, the host name is null. For example, for DRS urls like 'dos://dg.123/123-123-123',
      * even though 'dg.123' is a valid host, somehow since it does not conform to URI's standards, uri.getHost returns null. In such
-     * cases, authority is used instead of host.
+     * cases, authority is used instead of host. If there is no authority, use an empty string.
      */
     val uri = new URI(uriAsString)
     val host = uri.getHost
-    val hostOrAuthority = if (host == null) uri.getAuthority else host
-    require(!hostOrAuthority.isEmpty, s"Bucket/Host is empty")
+    val hostOrAuthorityOrEmpty =
```
Comment (cjllanwarne) [inline cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsCloudNioFileSystemProvider.scala]: Is this logic neater like this?
```
val hostOrAuthorityOrEmpty = 
  Option(uri.getHost).getOrElse(
    Option(uri.getAuthority).getOrElse("")
  )
```

File: filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilder.scala [hunk 2]
```diff
@@ -15,15 +15,12 @@ case class DrsPathBuilder(fileSystemProvider: DrsCloudNioFileSystemProvider) ext
 
   override def name: String = "DRS"
 
+  //DRS or DOS files can actually have no host, and treat that as valid (for now?)
```
Comment (cjllanwarne) [inline filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilder.scala]: TOL: Is this comment helpful here? TBH I'm not sure I'd know what to do with this information here.

**Comment by cjllanwarne** [comment]
It just occurred to me - this doesn't seem to do anything to allow both `drs` and `dos` URLs - I can't remember whether that was part of the previous version of this PR or some other PR and is already in develop?

**Comment by ruchim** [comment]
@cjllanwarne --it was part of my previous version but it isnt a requirement yet...and my implementation was rough, so I decided to descope based on current needs.

**Pull request merged**